include/alp/algorithms/cholesky.hpp

```diff
@@ -51,8 +52,16 @@ namespace alp {
 				is_matrix< MatL >::value &&
 				is_matrix< MatH >::value &&
 				structures::is_a< typename MatL::structure, structures::UpperTriangular >::value &&
-				// TODO: structures::Symmetric should be replced with structures::SymmetricPositiveDefinite
-				structures::is_a< typename MatH::structure, structures::Symmetric >::value &&
+				// TODO: structures::Symmetric should be replced
+				//       rewith structures::SymmetricPositiveDefinite
+				( (
+					!grb::utils::is_complex< D >::value &&
+					structures::is_a< typename MatH::structure, structures::Symmetric >::value
+				) ||
+				(
+					grb::utils::is_complex< D >::value &&
+					structures::is_a< typename MatH::structure, structures::Hermitian >::value
+				) ) &&
```

Comment:
I suggest to improve indentation because it is visually misleading to which `(` the `)` on line 60 belong to.